parsers/parsers.go

```diff
@@ -36,6 +45,35 @@ func (e *Empty) DoFromHuman(string) string {
 	return "Not Yet Implemented"
 }
 
-var ErrTooLarge error = errors.New("Too Beaucoup")
-var ErrTooSmall error = errors.New("Number too small")
-var ErrNotANumber error = errors.New("Not a Number")
+// IsMachineNumber validates that a number:
+// Contains only digits
+// The first digit must be 1-9 when >= 10
+func isMachineNumber(s string) bool {
+	match, _ := regexp.MatchString(`^[0-9]$|^[1-9][0-9]+$`, s)
+	if match {
+		return true
+	}
+	return false
+}
+
+// IsDelimitednumber validates that a number:
+// Contains only digits and` delimiters [., _]
+// The first digit must be 1-9 when >= 10
+// The first group can be 1-3 digits
+// Subsequent groups must be 3 digits
+func isDelimitedNumber(s string) bool {
+	// 0-999 are machine numbers
+	if len(s) < 4 {
+		if ok := isMachineNumber(s); ok {
+			return true
+		} else {
+			return false
+		}
+	}
+
+	match, _ := regexp.MatchString(`^[1-9][0-9]{0,2}([.,_ ][0-9]{3})+$`, s)
+	if match {
+		return true
+	}
+	return false
```

Comment:
to keep that line straight and reduce the code further we can leverage the bool returned by `regexp.MatchString` and return it directly removing the need for the `if` all together

```suggestion
	return match
```

Reply:
More info on line of sight in golang here https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88

I liked this talk